src/toast/ops/common_mode_noise.py

```diff
@@ -0,0 +1,241 @@
+            focalplane = obs.telescope.focalplane
+            fsample = focalplane.sample_rate.to_value(u.Hz)
+            freqs = np.linspace(0, fsample / 2, 10000)
+            freqs[0] = 1e-6
+
+            # Check that the noise model exists
+            if self.noise_model not in obs:
+                msg = (
+                    "Noise model {self.noise_model} does not exist in "
+                    "observation {obs.name}"
+                )
+                raise RuntimeError(msg)
+
+            noise = obs[self.noise_model]
+
+            # Find the unique values of focalplane keys
+
+            dets_by_key = {}
+            if self.focalplane_key is None:
+                dets_by_key[None] = []
+                for det in obs.all_detectors:
+                    if detset is not None and det not in detset:
+                        continue
+                    dets_by_key[None].append(det)
+            else:
+                if self.focalplane_key not in focalplane.detector_data.colnames:
+                    msg = f"Focalplane does not have column for '{self.focalplane_key}'"
+                    raise RuntimeError(msg)
+                for det in obs.all_detectors:
+                    key = focalplane[det][self.focalplane_key]
+                    if key not in dets_by_key:
+                        dets_by_key[key] = []
+                    dets_by_key[key].append(det)
+
+            # Create a new base-class noise object with the same PSDs and
+            # mixing matrix as the input.  Then modify those values.  If the
+            # output name is the same as the input, then delete the input
+            # and replace it with the new model.
+
+            nse_keys = noise.keys
+            nse_dets = noise.detectors
+            nse_freqs = {x: noise.freq(x) for x in nse_keys}
+            nse_psds = {x: noise.psd(x) for x in nse_keys}
+            nse_indx = {x: noise.index(x) for x in nse_keys}
+            mixing_matrix = noise.mixing_matrix
+
+            # Add the common mode noise PSDs
+
+            fmin = self.fmin.to_value(u.Hz)
+            fknee = self.fknee.to_value(u.Hz)
+            alpha = self.alpha
+            net = self.NET
+
+            if self.static_coupling:
+                obs_id = 0
+                realization = 0
+            else:
+                obs_id = obs.uid
+                realization = self.realization
+
+            for key, dets in dets_by_key.items():
+                if key is None:
+                    noise_key = f"{self.name}_{self.component}"
+                else:
+                    noise_key = f"{self.name}_{self.component}_{key}"
+                mixing_matrix[noise_key] = {}
+                noise_uid = name_UID(noise_key)
+                nse_keys.append(noise_key)
+                nse_freqs[noise_key] = freqs * u.Hz
+                nse_psds[noise_key] = net**2 * (
+                    (freqs**alpha + fknee**alpha) / (freqs**alpha + fmin**alpha)
+                )
```

Comment:
Not needed for this PR, but the analytic 1/f noise model [is used in several places](https://github.com/hpc4cmb/toast/blob/017d65603a3d6e6566090e52634fcd4965713de2/src/toast/noise_sim.py#L106-L114).  Perhaps we should make a helper function for that in `toast.sim_noise.py` someday?